python/tests/transformers/named_image_test.py

```diff
@@ -69,8 +69,9 @@ class NamedImageTransformerBaseTestCase(SparkDLTestCase):
     name = None
     # Allow subclasses to force number of partitions - a hack to avoid OOM issues
     numPartitionsOverride = None
-    featurizerCompareDigitsExact = 6
+    featurizerCompareDigitsExact = 5
     featurizerCompareDigitsCosine = 1
+    poolingMethod = None
```

Comment:
Is it true that pooling is None in testing and `avg` in prod? if so, can we clarify why in the description?

Reply:
For other models, we can use the default poolingMethod which is None. Only issue is ResNet50. The new Keras version changed the model. In order to make it match as what we did before, we need to add the avg pooling in ResNet50 model.

Reply:
1. I was looking at the test cases, can you point me to which one is it that fails?
If I understand correctly, we check the output of this model, as stored in `kerasPredict` against tensorflow, resized images and deepimagepredictor model. Shouldn't all of them be still the same? if not, which one and why?
2. also, should we simply add `pooling="avg"` in https://github.com/databricks/spark-deep-learning/blob/4a2ae917630dfc4aa7a9921d9810debb79991814/python/sparkdl/transformers/keras_applications.py#L228? to achieve the same effect as you are achieving without adding the testing harness to test against all poolings?

Reply:
The test `test_featurization_no_reshape` fails because kerasReshaped and dfFeatures has different shapes.

https://github.com/databricks/spark-deep-learning/blob/master/python/tests/transformers/named_image_test.py#L186

Reply:
About the second. My question is do we want to support other pooling method for Kreas model in the future?